### PR TITLE
fix watch resources in each controller

### PR DESCRIPTION
**Pull request by ldpliu**
## Summary
We did not watch globalhub applied resources previously,  if user changed these resources, we should rollback them.
## Related issue(s)

Fixes #

## Tests
* [ ] Unit/function tests have been added and incorporated into `make unit-tests`.
* [ ] Integration tests have been added and incorporated into `make integration-test`.
* [ ] E2E tests have been added and incorporated into `make e2e-test-all`.
* [x] List other manual tests you have done.
  - manually removed the manager role/rolebinding, they will be recreated.

### CODE REVIEW

File: operator/pkg/controllers/backup/backup_start.go [hunk 1]
```diff
@@ -69,10 +69,11 @@ func GetBackupController() *BackupReconciler {
 }
 
 func StartController(initOption config.ControllerOption) (config.ControllerInterface, error) {
-	log.Infof("start backup controller")
 	if backupController != nil {
 		return backupController, nil
 	}
+	log.Infof("start backup controller")
+
```
Comment (clyang82) [inline operator/pkg/controllers/backup/backup_start.go:76]: move it after `IsACMResourceReady`

File: operator/pkg/controllers/grafana/grafana_reconciler.go [hunk 2]
```diff
@@ -125,10 +126,11 @@ func (r *GrafanaReconciler) IsResourceRemoved() bool {
 }
 
 func StartController(initOption config.ControllerOption) (config.ControllerInterface, error) {
-	log.Info("start grafana controller")
 	if grafanaController != nil {
 		return grafanaController, nil
 	}
+	log.Info("start grafana controller")
```
Comment (clyang82) [inline operator/pkg/controllers/grafana/grafana_reconciler.go:132]: ditto

File: operator/pkg/controllers/inventory/inventory_reconciler.go [hunk 3]
```diff
@@ -68,13 +68,15 @@ func (r *InventoryReconciler) IsResourceRemoved() bool {
...
-	log.Info("start inventory controller")
 	if inventoryReconciler != nil {
 		return inventoryReconciler, nil
 	}
+
 	if !config.WithInventory(initOption.MulticlusterGlobalHub) {
 		return nil, nil
 	}
+	log.Info("start inventory controller")
+
```
Comment (clyang82) [inline operator/pkg/controllers/inventory/inventory_reconciler.go:79]: move it after `GetStorageConnection`

File: operator/pkg/controllers/grafana/grafana_reconciler.go [hunk 4]
```diff
@@ -156,7 +158,17 @@ func (r *GrafanaReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Watches(&corev1.ConfigMap{},
 			&handler.EnqueueRequestForObject{}, builder.WithPredicates(configmapPred)).
 		Watches(&appsv1.Deployment{},
-			&handler.EnqueueRequestForObject{}, builder.WithPredicates(deplomentPred))
+			&handler.EnqueueRequestForObject{}, builder.WithPredicates(deploymentPred)).
+		Watches(&corev1.Service{},
```
Comment (clyang82) [inline operator/pkg/controllers/grafana/grafana_reconciler.go:162]: Need to add more permissions for this controller

File: operator/pkg/controllers/managedhub/managedhub_controller.go [hunk 5]
```diff
@@ -55,10 +55,11 @@ func (r *ManagedHubController) IsResourceRemoved() bool {
 }
 
 func StartController(initOption config.ControllerOption) (config.ControllerInterface, error) {
-	log.Info("start managedhub controller")
 	if managedHubController != nil {
 		return managedHubController, nil
 	}
+	log.Info("start managedhub controller")
+
```
Comment (clyang82) [inline operator/pkg/controllers/managedhub/managedhub_controller.go:62]: move it after `IsACMResourceReady`

File: operator/pkg/controllers/manager/manager_reconciler.go [hunk 6]
```diff
@@ -115,12 +118,32 @@ func (r *ManagerReconciler) IsResourceRemoved() bool {
...
+			&handler.EnqueueRequestForObject{}, builder.WithPredicates(config.GeneralPredicate)).
+		Watches(&corev1.ServiceAccount{},
+			&handler.EnqueueRequestForObject{}, builder.WithPredicates(config.GeneralPredicate)).
+		Watches(&rbacv1.ClusterRole{},
+			&handler.EnqueueRequestForObject{}, builder.WithPredicates(config.GeneralPredicate)).
+		Watches(&rbacv1.ClusterRoleBinding{},
+			&handler.EnqueueRequestForObject{}, builder.WithPredicates(config.GeneralPredicate)).
+		Watches(&rbacv1.Role{},
+			&handler.EnqueueRequestForObject{}, builder.WithPredicates(config.GeneralPredicate)).
+		Watches(&rbacv1.RoleBinding{},
```
Comment (clyang82) [inline operator/pkg/controllers/manager/manager_reconciler.go:136]: add more permissions for this controller

File: operator/pkg/controllers/storage/storage_reconciler.go [hunk 7]
```diff
@@ -101,26 +99,38 @@ func StartController(initOption config.ControllerOption) (config.ControllerInter
...
+			&handler.EnqueueRequestForObject{}, builder.WithPredicates(config.GeneralPredicate)).
 		Watches(&appsv1.StatefulSet{},
 			&handler.EnqueueRequestForObject{}, builder.WithPredicates(statefulSetPred)).
-		Complete(r)
+		Watches(&corev1.ServiceAccount{},
+			&handler.EnqueueRequestForObject{}, builder.WithPredicates(config.GeneralPredicate))
+
+	if r.enableMetrics {
+		mgrBuilder = mgrBuilder.
+			Watches(&promv1.PrometheusRule{},
```
Comment (clyang82) [inline operator/pkg/controllers/storage/storage_reconciler.go:127]: Does it work when switch enableMetrics from false to true?

File: operator/pkg/controllers/transporter/protocol/strimzi_kafka_controller.go [hunk 8]
```diff
@@ -148,7 +148,7 @@ var kafkaPred = predicate.Funcs{
 }
 
 func StartKafkaController(ctx context.Context, mgr ctrl.Manager, transporter transport.Transporter) error {
-	log.Info("start kafka controller")
+	log.Debug("start kafka controller")
 	if startedKafkaController {
 		return nil
 	}
```
Comment (clyang82) [inline operator/pkg/controllers/transporter/protocol/strimzi_kafka_controller.go:154]: Move it after `startedKafkaController`. Use `Info` for consistence?

File: operator/pkg/controllers/webhook/webhook_controller.go [hunk 9]
```diff
@@ -59,10 +61,15 @@ func NewWebhookReconciler(mgr ctrl.Manager,
 }
 
 func StartController(opts config.ControllerOption) (config.ControllerInterface, error) {
-	log.Info("start webhook controller")
 	if webhookReconciler != nil {
 		return webhookReconciler, nil
 	}
+	log.Info("start webhook controller")
```
Comment (clyang82) [inline operator/pkg/controllers/webhook/webhook_controller.go:67]: move it after `IsACMResourceReady`

**Review by clyang82 — CHANGES_REQUESTED** [review]

File: operator/pkg/controllers/backup/backup_start.go (hunk 1 above)
Comment (ldpliu) [reply operator/pkg/controllers/backup/backup_start.go]: Done

File: operator/pkg/controllers/grafana/grafana_reconciler.go (hunk 2 above)
Comment (ldpliu) [reply operator/pkg/controllers/grafana/grafana_reconciler.go]: Done

File: operator/pkg/controllers/inventory/inventory_reconciler.go (hunk 3 above)
Comment (ldpliu) [reply operator/pkg/controllers/inventory/inventory_reconciler.go]: Done

File: operator/pkg/controllers/grafana/grafana_reconciler.go (hunk 4 above)
Comment (ldpliu) [reply operator/pkg/controllers/grafana/grafana_reconciler.go:163]: Done

File: operator/pkg/controllers/managedhub/managedhub_controller.go (hunk 5 above)
Comment (ldpliu) [reply operator/pkg/controllers/managedhub/managedhub_controller.go]: Done

File: operator/pkg/controllers/storage/storage_reconciler.go (hunk 7 above)
Comment (ldpliu) [reply operator/pkg/controllers/storage/storage_reconciler.go]: wath the resources when crd exist

File: operator/pkg/controllers/transporter/protocol/strimzi_kafka_controller.go (hunk 8 above)
Comment (ldpliu) [reply operator/pkg/controllers/transporter/protocol/strimzi_kafka_controller.go:153]: Done

File: operator/pkg/controllers/webhook/webhook_controller.go (hunk 9 above)
Comment (ldpliu) [reply operator/pkg/controllers/webhook/webhook_controller.go]: Done

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**